adapter-neo4j/src/main/java/io/nosqlbench/adapter/neo4j/Neo4JAdapterUtils.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024 nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.nosqlbench.adapter.neo4j;
+
+import org.apache.commons.lang3.StringUtils;
+
+import org.neo4j.driver.Record;
+import org.neo4j.driver.exceptions.ClientException;
+
+import java.util.NoSuchElementException;
+
+
+public class Neo4JAdapterUtils {
+
+    /**
+     * Mask the digits in the given string with '*'
+     *
+     * @param unmasked The string to mask
+     * @return The masked string
+     */
+    protected static String maskDigits(String unmasked) {
+        assert StringUtils.isNotBlank(unmasked) && StringUtils.isNotEmpty(unmasked);
+        int inputLength = unmasked.length();
+        StringBuilder masked = new StringBuilder(inputLength);
+        for (char ch : unmasked.toCharArray()) {
+            if (Character.isDigit(ch)) {
+                masked.append("*");
+            } else {
+                masked.append(ch);
+            }
+        }
+        return masked.toString();
+    }
+
+    /**
+     * Reference:
+     * - https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/Value.html#asObject()
+     */
+    public static Object[] getFieldForAllRecords(Record[] records, String fieldName) {
+        int n = records.length;
+        Object[] values = new Object[n];
+        int idx;
+        for (int i = 0; i < n; i++) {
+            try {
+                idx = records[i].index(fieldName);
+                values[i] = records[i].get(idx).asObject();
```

Comment:
If any one record does not have the required field, then the call to this method fails for all and exits the loop throwing an exception. Just verifying that this is the desired behavior as opposed to eg setting values[i] for that i to null or something and continuing iteration.

Reply:
right, this was the intended behavior for this parsing of the `Record` array. 

I figured if the user is leveraging this method, they are actually expecting the field from all `Record`s that Neo4J returned to them. So if we throw an exception upon not getting the field, and this forces the user to go back and check their data and/or data model.

Should we add a `tryToGetFieldForallRecords` method here, that just populates an index in values with `null` if it's not found?